lib/declarative_authorization/authorization.rb

```diff
@@ -340,11 +340,16 @@ def user_roles_privleges_from_options(privilege, options)
       raise AuthorizationUsageError, "No user object given for user id (#{user.try(:id)}) or " +
         "set through Authorization.current_user" unless user
 
-      roles = options[:user_roles] || flatten_roles(roles_for(user))
+      roles = options[:user_roles] || flattened_roles(user)
       privileges = flatten_privileges privileges, options[:context]
       [user, roles, privileges]
     end
 
+    def flattened_roles(user)
```

Comment:
I don't remember if there are use cases that may change these roles for a user. If so, do we need to invalidate this cache in those cases? Or, is this cache you are creating short-lived, i.e. per-request?

Reply:
I think the hash only exists per request, so if someone changes user role for a user, the next page load would be in a different request, which would calculate this again.